File: frontend/js/create_product.js
$(document).on("click", "#createProduct", function() {
    let name = $("#nameField").val()
    let description = $("#descriptionField").val()
    let perishable = booleanFixer($("#perishableField").val())
    let freezable = booleanFixer($("#freezableField").val())
    let price = parseFloat($("#priceField").val())
    let expired = booleanFixer($("#expiredField").val())
    let validity = $("#validityField").val()

    if (validity != "") {
        data = JSON.stringify({name: name, description: description,
        is_perishable: perishable, freezable: freezable, price: price, 
        expired: expired, validity: validity})
    }
    else {
        data = JSON.stringify({name: name, description: description,
        is_perishable: perishable, freezable: freezable, price: price,
        expired: expired, validity: null})
    }
    
    ajaxFunction("new_product", "POST", data, function() {
        alert("The product has been created!")
    })
})

File: routes/create_product.py
from general.config import *
from models.product import *
from general.tools import *
from models.person import *

@app.route("/new_product", methods= ["POST"])
@jwt_required()
def new_product():
    
    current_user = Person.query.filter(Person.email == get_jwt_identity()).first()
    user_permission = current_user.permission.name
    if can_do("create_product", "post", user_permission):
        data = request.get_json()
        try:
            if data["validity"]:
                data["validity"] = date_format(data["validity"])
            product = Product(**data)
            db.session.add(product)
            db.session.commit()
            result = result_generator("product", to_json(product))
        except Exception as e:
            result = new_result("error", str(e))
    else:
        result = new_result("error", "insufficient permission")
    result.headers.add("Access-Control-Allow-Origin", "*")
    return result

# curl -d '{"name": "orange", "description": "orange fruit", "is_perishable": true, "freezable": false, "price": 0.40, "expired": false, "validity": "2023-01-17"}' -X POST -H "Content-Type:application/json" localhost:5000/new_product